make-util.js

```diff
@@ -1203,9 +1205,233 @@ var createNonAggregatedZip = function (buildPath, packagePath) {
     console.log('> Zipping non-aggregated tasks layout');
     var nonAggregatedZipPath = path.join(packagePath, 'non-aggregated-tasks.zip');
     compressTasks(nonAggregatedLayoutPath, nonAggregatedZipPath);
+
+    return nonAggregatedLayoutPath;
 }
 exports.createNonAggregatedZip = createNonAggregatedZip;
 
+/**
+ * Create a NuGet package per task. This function assumes the tasks are already laid out on disk.
+ * 
+ * When running locally, layoutPath is something like: _package\non-aggregated-layout
+ * Within this folder we have one of these folders per task:
+ *  /CmdLineV2
+ *      /Strings
+ *      /task.json
+ *      /task.loc.json
+ *      /task.zip
+ * 
+ * Within the function we create an artifacts folder, this is what gets uploaded when we are done.
+ * The contents look something like:
+ * /artifacts
+ *  /AndroidSigningV2
+ *      /Mseng.MS.TF.DistributedTask.Tasks.AndroidSigningV2.2.135.0.nupkg
+ *      /push.cmd
+ *  /AnotherTask
+ *      /Mseng.MS.TF.DistributedTask.Tasks.AnotherTaskV1.1.0.0.nupkg
+ *      /push.cmd
+ *  /push.cmd * Root push.cmd that runs all nested push.cmd's.
+ *  /servicing.xml * Convenience file. Generates all XML to update servicing configuration for tasks.
+ *  /unified_deps.xml * Convenience file. Generates all XML to update unified dependencies file.
+ * 
+ * @param {*} packagePath Path of _packages folder.
+ * @param {*} layoutPath Path that has task layouts.
+ */
+var createNugetPackagePerTask = function (packagePath, /*nonAggregatedLayoutPath*/layoutPath) {
+    console.log();
+    console.log('> Creating NuGet package per task')
+
+    // create folder for _package\task-zips
+    // Inside this folder we have one folder per task that contains the task.zip.
+    // It also has layout-version.txt and the nuspec file for the task.
+    // This folder is what we create the nupkg from.
+    console.log();
+    console.log('> Creating task zips folder');
+    var tasksZipsPath = path.join(packagePath, 'task-zips');
+    mkdir('-p', tasksZipsPath);
+
+    // _package\artifacts
+    // This is the final state of the task content and what is published as a build artifact.
+    console.log();
+    console.log('> Creating artifacts folder');
+    var artifactsPath = path.join(packagePath, "artifacts");
+    mkdir('-p', artifactsPath);
+
+    console.log();
+    console.log('> Zipping task folders')
+
+    // maintain package references that we need to add to unified dependencies
+    var unifiedDepsContent = [];
```

Comment:
I am keeping these for now as it makes it much easier to create the xml config files we need in VSO. We can remove once it's fully rolled out.